packages/kit/types/index.d.ts

```diff
@@ -253,7 +253,9 @@ export interface RequestEvent<
  *
  * It receives `Params` as the first generic argument, which you can skip by using [generated types](/docs/types#generated-types) instead.
  */
-export interface RequestHandler<Params extends Record<string, string> = Record<string, string>> {
+export interface RequestHandler<
```

Comment:
@dummdidumm shouldn't this be
```ts
export interface RequestHandler<
	Params extends Record<string, string> = Record<string, string>
> {
	(event: RequestEvent<Partial<Params>>): MaybePromise<Response>;
}
```
?

Reply:
Why exactly?

Reply:
I assume you want to makr all `params` of the `event` as optional 

[Here is an example](https://www.typescriptlang.org/play?#code/PTAEGMFcCdoUwHYBdQEsC2AHANndikBDJVAewQFgAoOAD01OhVWTmgDNDw5QANARgA8ABULRC6AM6g6SRABNpopqkLZBAJTjhG8wZKTQWAcwA0oA0YTGAfDdABeUMpJrN23fsMnzlk3fsAb2pQUNAACkwxCUkALmdoqQBKeIA3UlR5agBfamodBANQWn54gUFA0CjxdFKLb2tzaokAJgB+eL9rUGz7JwRIbGxQQmlCBABPPKoS8Mrm2viAciWepNAQUABJCHHQUkx0VBQAAwWWk+pZwOz1zZ2EODh5UCRSKtHpM8T+S6ppzbsVC0ah0BhMNCsDhcHi8FqCEJhZQxGS0OQIRSgLQ6aB6LpmepWWyOLEeXFeIm+Bq2RGhIK0iILOIJFRuZFSGwpUDpTI5aYFIq0FpleHzH6dak9PqgAZDEZjSbTIVzD41OorNYbMA7cB7A5HU7nP7Km53bWgeTkJYoR7PV7vKKSaRM6ZAA) that demonstrates the issue:

Reply:
I see. No it's working as intended. What you pass for the generic should be applied as-is, i.e. required params should be passed and can be expected to exist. It changes to `Partial<..>` so that in case you don't type the generic you can't just access the park without checking if it exists (in strict mode)

Reply:
Ah, now I get it. Thanks for the explanation. Didn't knew this trick before.